experimental/CliffordAlgOrd/src/CliffordOrder.jl

```diff
@@ -0,0 +1,1058 @@
+end
+
+function (C::CliffordAlgebra)(elt::ZZCliffordOrderElem)
+  @req C === algebra(parent(elt)) "The Clifford algebra provided and the
+    ambient algebra of the Clifford order containing the element provided are not
+    identical."
+  return C(coefficients(elt))
+end
+
+function (C::ZZCliffordOrder)(elt::CliffordAlgebraElem)
+  @req algebra(C) === parent(elt) "The ambient algebra of the Clifford order
+    provided and the parent object of the element provided are not identical."
+    return C(coefficients(elt))
+end
+
+################################################################################
+#
+#  Containment
+#
+################################################################################
+
+@doc raw"""
+    in(x::CliffordAlgebraElem, C::CliffordOrder)
+
+Return true if the element $x$ is contained in the Clifford order $C$.
+"""
+function Base.in(x::CliffordAlgebraElem, C::CliffordOrder)
+  if !(algebra(C) === parent(x))
+    return false
+  end
+  coe, coeids = coefficients(x), coefficient_ideals(C)
+  for i in 1:rank(C)
+    if !(coe[i] in coeids[i])
+      return false
+    end
+  end
+  return true 
+end
+
+### ZZ ###
+
+@doc raw"""
+    in(x::CliffordAlgebraElem, C::ZZCliffordOrder)
+
+Return true if the element $x$ is contained in the Clifford order $C$.
+"""
+function Base.in(x::CliffordAlgebraElem, C::ZZCliffordOrder)
+  if !(algebra(C) === parent(x))
+    return false
+  end
+  coe = coefficients(x)
+  for i in 1:rank(C)
+    if !(is_integral(coe[i]))
+      return false
+    end
+  end
+  true 
+end
+
+################################################################################
+#
+#  Basic field access
+#
+################################################################################
+
+##### Order  #####
+@doc raw"""
+    base_ring(C::CliffordOrder) -> Ring
+
+Return the base ring of the Clifford order $C$.
+"""
+base_ring(C::CliffordOrder) = C.base_ring::base_ring_type(typeof(C))
+
+@doc raw"""
+    algebra(C::CliffordOrder) -> CliffordAlgebra
+
+Return the Clifford algebra of the ambient space of the underlying lattice of $C$.
+"""
+algebra(C::CliffordOrder) = C.algebra
+
+@doc raw"""
+    rank(C::CliffordOrder) -> Int
+
+Return the rank of the Clifford order $C$ over its base ring.
+"""
+rank(C::CliffordOrder) = C.rank
+
+@doc raw"""
+    lattice(C::CliffordOrder) -> QuadLat
+
+Return the underlying even quadratic lattice of the Clifford order $C$.
+"""
+lattice(C::CliffordOrder) = C.lattice
+
+@doc raw"""
+    gram_matrix(C::CliffordOrder) -> MatElem
+
+Return the Gram matrix with respect to the fixed pseudo-basis of the Clifford order $C$.
+"""
+gram_matrix(C::CliffordOrder) = C.gram
+
+@doc raw"""
+    coefficient_ideals(C::CliffordOrder) -> Vector{NumFieldOrderFractionalIdeal}
+
+Return the vector of coefficient ideals of the canonical pseudo-basis of $C$.
+"""
+coefficient_ideals(C::CliffordOrder) = C.coefficient_ideals::Vector{typeof(fractional_ideal(base_ring(C), one(base_ring(C))))}
```

Comment:
I believe there is fractional_ideal_type to get this type at compile time (ie without typeof). Maybe you need to write Hecke.fractional_ideal_type (not sure, maybe @thofma knows)

Reply:
`Hecke.fractional_ideal_type(base_ring(C))` should to the trick.

Reply:
Even better to use `Hecke.fractional_ideal_type(base_ring_type(C))` to make it compile time computable

Reply:
Yes, this works